src/apps/chifra/pkg/output/json_writer.go

```diff
@@ -0,0 +1,332 @@
+package output
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/rpcClient"
+)
+
+type FieldType int
+
+var comma = ","
+
+const (
+	FieldArray FieldType = iota
+	FieldObject
+)
+
+type position int
+
+const (
+	positionEmpty position = iota
+	positionInRoot
+	positionInArray
+	positionInObject
+	positionRootClosed
+)
+
+type state = struct {
+	position
+	children int
+}
+
+type DefaultField struct {
+	Key string
+	FieldType
+}
+
+// JsonWriter can write JSON object in portions.
+type JsonWriter struct {
+	// current state of the JSON object (are we inside of
+	// an array or object? are there any children?)
+	state state
+	// previous state is helpful when closing an object or
+	// array, so we will keep it as a stack
+	previousStates []state
+	// the writer that we will output to
+	outputWriter io.Writer
+	indentLevel  int
+	indentString string
+	// errors to output
+	errs []string
+	// function to get meta data
+	GetMeta func() (*rpcClient.MetaData, error)
+	// flag indicating if we should output `meta` object as
+	// well
+	ShouldWriteMeta bool
+	// Should writer write newline after `Close`
```

Comment:
Maybe this is always `true`? Is the only reason for this in order to agree with test cases, or is there some other reason?